cmd/parse-genesis/parseGenesis.go

```diff
@@ -0,0 +1,45 @@
+package parsegenesis
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/forbole/bdjuno/v2/cmd/parse-genesis/utils"
+	"github.com/forbole/juno/v2/cmd/parse"
+	"github.com/forbole/juno/v2/types/config"
+	junoutils "github.com/forbole/juno/v2/types/utils"
+	"github.com/spf13/cobra"
+)
+
+// NewParseGenesisCmd returns the Cobra command allowing to parse the genesis file
+func NewParseGenesisCmd(parseCfg *parse.Config) *cobra.Command {
+	return &cobra.Command{
+		Use:     "parse-genesis [optional: module names]",
+		Short:   "Parse genesis file. To parse specific modules, input module names as arguments",
+		Example: "bdjuno parse-genesis auth bank consensus gov history staking",
+		PreRunE: parse.ReadConfig(parseCfg),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			parseCtx, err := parse.GetParsingContext(parseCfg)
+			if err != nil {
+				return err
+			}
+
+			genesisDoc, genesisState, err := junoutils.GetGenesisDocAndState(config.Cfg.Parser.GenesisFilePath, parseCtx.Node)
+			if err != nil {
+				return fmt.Errorf("error while getting genesis doc or state: %s", err)
+			}
+
+			invalidInputs, err := utils.ParseGenesis(parseCtx.Modules, genesisDoc, genesisState, args)
+			if err != nil {
+				return fmt.Errorf("error while parsing genesis: %s", err)
+			}
```

Comment:
Instead of using this logic, I think a better approach could be the following: 

1. Take the list of modules to parse: if provided by the user, use those. Otherwise use the default list 
2. Iterate through all the specified modules, and find anyone that does not match. For each module name:
   - if found, add that module to a list of modules to parse 
   - if not found, return an error
3. Once we have validated all the modules names, simply go through all the modules to parse and call `HandleGenesis`

All of this can be done inside this method rather than defining another method

Reply:
cool, I have made the changes. As for the module not found, should we return error or just print the log that the module is not registered?

Reply:
@huichiaotsou I think it's better to return an error here since it's something that is specified by hand from the user and they should check out whether it was a typo or an error inside the config